client/protocols/export.py

```diff
@@ -0,0 +1,145 @@
+    """Downloads submissions from the server"""
+
+    def on_interact(self):
+        """If the --export parameter was used, downloads all submissions 
+        for the given assignment from the server and then exits
+        """
+        data = None
+        try:
+            data = pickle.load(open("export_cache.pkl", "rb"))
+            self.access_token = auth.authenticate(self.args.authenticate)
+            log.info('Authenticated with access token %s', self.access_token)
+        except Exception as e:
+            if self.args.export:
+                data = {}
+                self.access_token = auth.authenticate(self.args.authenticate)
+                log.info('Authenticated with access token %s', self.access_token)
+                data['assign'] = self.args.export
+                data['course'] = self.get_course()
+                data['students'] = self.get_students(data['course'])
+                data['assignid'] = self.get_assign_id(
+                    data['course'], data['assign'])
+                data['current'] = 0
+        
+        if data:
+            try:
+                if not os.path.exists(data['assign']): 
+                    os.makedirs(data['assign'])
+                
+                downloads_left = len(data['students']) - data['current']
+                
+                print("{0} submissions left to download".format(downloads_left))
+                
+                for i in range(data['current'], len(data['students'])):
+                    student = data['students'][i]
+                    try:
+                        self.download_submission(student, data['assign'], 
+                            data['assignid'])
+                        data['current'] = i + 1
+                    except Exception as e:
+                        data['current'] = i
+                        pickle.dump(data, open("export_cache.pkl", "wb"))
+                        dl_left = len(data['students']) - i
+                        print("Download failed. Run command again to continue.")
+                        print("{0} submissions left to download".format(dl_left))
+                        sys.exit(0)
+                        
+            except KeyboardInterrupt:
+                pickle.dump(data, open("export_cache.pkl", "wb"))
+                dl_left = len(data['students']) - i
+                print("Download interrupted. Run command again to continue.")
+                print("{0} submissions left to download".format(dl_left))
+                sys.exit(0)
+                
+            print("Submissions downloaded.")
+            if os.path.exists('export_cache.pkl'):
+                os.remove('export_cache.pkl')
+            sys.exit(0)
+            
+    def get_course(self):
+        """Gets a course ID to search for"""
+        if self.args.course > 0:
+            return self.args.course
+        print("Loading course list...")
+        response = self.request('course')
+        num = 0
```

Comment:
Since this variable isn't used in the `if` or the `elif`, you should move it right before the `for` loop.